### DISCUSSION TITLE
USB Speed

### DISCUSSION THREAD

**Discussion by joelsernamoreno** [adafruit/Adafruit_TinyUSB_Arduino#409] [Q&A]
Hello!

By default RP2040 is a USB Full speed device

Is it possible to modify RP2040 to be a USB LOW speed device? Where can I modify this?

Thanks

**Comment by hathach** [comment]
have you checked rp2040 datasheet to find way to force low speed link ? Then you just modify the stack whenever it needs to be modified

**Reply by joelsernamoreno** [reply]
Hi @hathach 

Thank you for your response. 

I have looked at the rp2040 datasheet but I don't quite understand how to configure this as USB low speed

I have modified the number of maximum endpoints from 64 to 8 (I have seen this in another question from someone else), but this is not enough.

Would you be so kind to provide some more information?

Thank you!
